tree/ntuple/v7/test/ntuple_types.cxx

```diff
@@ -14,7 +14,11 @@ TEST(RNTuple, TypeName)
 {
    EXPECT_STREQ("float", ROOT::Experimental::RField<float>::TypeName().c_str());
    EXPECT_STREQ("std::vector<std::string>", ROOT::Experimental::RField<std::vector<std::string>>::TypeName().c_str());
+#if defined(_MSC_VER) && !defined(R__ENABLE_BROKEN_WIN_TESTS)
+   EXPECT_STREQ("struct CustomStruct", ROOT::Experimental::RField<CustomStruct>::TypeName().c_str());
```

Comment:
Interesting!  So `ROOT::Internal::GetDemangledTypeName(typeid(CustomStruct))` returns a different string on Windows. Is this a new behavior?  In any cast, it will break RNTuple if left unhandled, so the test fails for good.  I need to look into it.  If it is urgent, we can comment the test and add a `TODO`.

Reply:
The `R__ENABLE_BROKEN_WIN_TESTS` is there to run the broken tests. And obviously, the unmangled (undecorated) names are different on Windows than on Linux. But in this particular case, I don't know if this is a bug or a feature... Maybe @Axel-Naumann can have an idea?

Reply:
On Windows, `struct`s and `class`es can have different ABIs, so it might be intentional given that it's part of the mangling. Now - is that what *we* want? Maybe not, so it might be a good idea to fix `ROOT::Internal::GetDemangledTypeName`. Either way: if this gets merged, can I have a bug report such that we don't forget about this demangling issue?

Reply:
OK, I'll open a bug report. There is also at least one more test (in roottest) where I had to change the ref file to match the correct demangled names...

Reply:
@jblomer @Axel-Naumann I have created the [Jira issue ROOT-10794](https://sft.its.cern.ch/jira/browse/ROOT-10794). Can I merge it, or do you wand to do it differently?

Reply:
Please go ahead and merge.  I'll follow up on the RNTuple side separately.

Reply:
OK, thanks Jakob! Merging.